index.js

```diff
@@ -16,8 +18,12 @@ function makeSubject() {
         sink = zinkz[i];
         if (sinks.indexOf(sink) > -1) sink(type, data);
       }
+      if (type === 2) {
+        done = true;
+        sinks.length = 0;
```

Comment:
Could we not have a mutation of the array `length` property? Seems very unconventional to do such mutation of a property we don't own, and I also think we don't need it here to achieve #5.

Reply:
how do we not own it? we own the list and the `length` property is, [per spec, a read/write property](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length). I suspect this is the fastest way to clear it out (though `sinks.splice(0, sinks.length))` should work equally well to the best of my knowledge).

for needing it: yeah its just being super safe on maintained references. I suspect (but I might be mistaken) that if for example someone holds a reference to the subject after completing it and we do not clear `sinks`, then references to all sinks who did not terminate on their own will also be maintained unnecessarily.